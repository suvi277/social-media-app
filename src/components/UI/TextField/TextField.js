import React from 'react';
import './TextField.scss';

const TextField = ({ name, type, placeholder, onChange, className, value, error, children, label, ...props }) => {
	return (
		<div className="form-group">
			<label htmlFor={name}>{label}</label>
			<input
				id={name}
				name={name}
				type={type}
				placeholder={placeholder}
				onChange={onChange}
				value={value}
				className={`form-control ${className}`}
				style={error && { border: 'solid 1px red' }}
			/>
			{error && <p>{error}</p>}
		</div>
	);
};

export { TextField };
