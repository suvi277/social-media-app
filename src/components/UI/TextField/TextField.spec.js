import React from 'react';
import { shallow } from 'enzyme';
import { TextField } from './TextField';

describe('TextField Component', () => {
	const wrapper = shallow(<TextField />);

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	//To Complet Test Cases
});
