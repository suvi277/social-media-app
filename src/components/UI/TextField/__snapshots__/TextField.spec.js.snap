// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Component renders as expected 1`] = `ShallowWrapper {}`;
